code/tests/unittests/test_vector3.cpp

```diff
@@ -156,3 +144,19 @@ TEST_F(Vector3Test, LengthSquaredWhenNoParamsReturnsLengthSquaredValueOfTheVecto
 
     ASSERT_THAT(actualValue, expectedValue);
 }
+
+TEST_P(Vector3ParameterizedTest, BracketOperatorWhenPassingAnIndexReturnsDoubleInThatPosition) {
+    double actual_value= test_vector[GetParam()];
+
+    ASSERT_EQ(actual_value, 1.0);
```

Comment:
What do you think about using a vector with different coordinates, such as (1, 2, 3), so that we make it harder to have false positives. For example, what if we had a vector v(1, 1, 1) and there's a bug that causes `v[2]` return `v.e[0]`? The assertion `ASSERT_EQ(actual_value, 1.0)` would be true, but the `Vector3` class is buggy. Instead, if we use a vector with all different coordinates, then it is harder to not catch a bug like the one I described.

I'm going to push some changes to the branch with my suggestion. If you disagree, we can revert them :)